pkg/quota/evaluator/core/pods.go

```diff
@@ -380,7 +380,10 @@ func podMatchesSelector(pod *api.Pod, selector api.ScopedResourceSelectorRequire
 	if err != nil {
 		return false, fmt.Errorf("failed to parse and convert selector: %v", err)
 	}
-	m := map[string]string{string(api.ResourceQuotaScopePriorityClass): pod.Spec.PriorityClassName}
+	var m map[string]string
+	if pod.Spec.PriorityClassName != "" {
```

Comment:
@k82cn in the "quota by priority" logic today there is no special handling for any priority. It does not matter from quota perspective if priority in the pod is "system-cluster-critical" , default or any other. What matters is if priority in quota's scopes matches pod's priority.

Reply:
What I means is priority class name maybe updated in other admission controller, how resource quota admission controller handle that?

Reply:
LGTM as we slacked; opened https://github.com/kubernetes/kubernetes/pull/66854 to handled "but did not set priority class name to default priority class name" case.

Reply:
I think you could just use a `len(pod.Spec.PriorityClassName) > 0` condition here instead of checking for empty string and that should eliminate the need for having empty string as priorityClass.

Reply:
@ravisantoshgudimetla Thanks! done!!

Reply:
Thanks @vikaschoudhary16. @k82cn do we still need the PR you introduced with this change? Either way this PR lgtm.